scripts/test

```diff
@@ -1,15 +1,25 @@
 #!/bin/bash
 set -e
 
-cd $(dirname $0)/..
+TESTS=${TESTS:-""}
 
-args=$1
+cd "$(dirname "$0")/.."
 
-echo Running tests
+args=("$@")
 
-PACKAGES="$(find . -name '*.go' | xargs -I{} dirname {} | sort -u | grep -Ev '(.git|.trash-cache|vendor|bin|k8s)')"
+echo "Running tests ${TESTS}"
 
-echo Packages: ${PACKAGES}
+if ! git diff --name-only | grep -Ev '(.git|.trash-cache|vendor|bin|k8s)' | grep -E '.*\.go' ; then
+  PACKAGES="$(find . -name '*.go' -exec dirname {} \; | sort -u | grep -Ev '(.git|.trash-cache|vendor|bin|k8s)')"
+else
+  PACKAGES="$(git diff --name-only | grep -E '.*\.go' | xargs -I{} dirname ./{} | sort -u | grep -Ev '(.git|.trash-cache|vendor|bin|k8s)')"
+fi
+
+echo Packages: "${PACKAGES}"
 
 [ "${ARCH}" == "amd64" ] && RACE=-race
-go test ${RACE} ${args} ${PACKAGES}
+if [ -n "${TESTS}" ]; then
+  go test -v "${RACE}" "${args[@]}" ${PACKAGES} -check.f "${TESTS}"
+else
+  go test -v "${RACE}" "${args[@]}" ${PACKAGES}
```

Comment:
Did you mean to also wrap "${PACKAGES}" in quotes as well, for the same reasons?  It make shellcheck happy.

Reply:
It's deliberately not quoted to let the shell re-split the words. The "correct" way would be to generate an array of package names in `$PACKAGES` and use `"${PACKAGES[@]}"` with quotes to pass the package names individually without expanding globs. However this is way more involved and I can pretty safely assume that we're not going to introduce package names with globs in them in the near future.
If this was quoted as you suggest, the shell would pass a single string of all package names to `go test` instead of individual strings with each package name individually. This would not work.